src/main/java/appeng/client/gui/implementations/GuiCraftConfirm.java

```diff
@@ -390,17 +435,29 @@ public void drawFG( final int offsetX, final int offsetY, final int mouseX, fina
 	@Override
 	public void drawBG( final int offsetX, final int offsetY, final int mouseX, final int mouseY )
 	{
-		this.setScrollBar();
-		this.bindTexture( "guis/craftingreport.png" );
-		this.drawTexturedModalRect( offsetX, offsetY, 0, 0, this.xSize, this.ySize );
+		this.bindTexture( this.getBackground() );
+		final int x_width = 238;
+		this.drawTexturedModalRect( offsetX, offsetY, 0, 0, x_width, 19 );
+
+		for( int x = 0; x < this.rows; x++ )
+		{
+			this.drawTexturedModalRect( offsetX, offsetY + 19 + x * SECTION_HEIGHT, 0, 19, x_width, SECTION_HEIGHT );
+		}
+
+		this.drawTexturedModalRect( offsetX, offsetY + 19 + this.rows * SECTION_HEIGHT - 1, 0, 133, x_width, 72 );
+	}
+
+	protected String getBackground()
+	{
+		return "guis/craftingreport.png";
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this method and declare a constant for this value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS3400)